flink-runtime/src/main/java/org/apache/flink/runtime/operators/Driver.java

```diff
@@ -87,4 +87,6 @@
 	 * @throws Exception Exceptions may be forwarded.
 	 */
 	void cancel() throws Exception;
+
+	void resetForIterativeTasks() throws Exception;
```

Comment:
> You mean all drivers will now become a type of ResettableDriver?

Yes, this was the idea in my comment [1] on the Jira. I'm sorry if I wasn't clear enough there.
The advantage of this is that you wouldn't need to add more logic in `BatchTask`, `AbstractIterativeTask` and its descendants, because these already handle the situation when the driver is an instance of `ResettableDriver`. (See e.g. the beginning of `reinstantiateDriver`.)

[1] https://issues.apache.org/jira/browse/FLINK-3322?focusedCommentId=15474531&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15474531